bluemira/power_cycle/base.py

```diff
@@ -4,23 +4,64 @@
 Base classes for the power cycle model.
 """
 from abc import ABC, ABCMeta, abstractmethod, abstractproperty
-from typing import Union
+from dataclasses import dataclass
+from enum import Enum, auto
+from typing import Any, Iterable, Union
 
 import numpy as np
+from numpy.typing import ArrayLike
 
-from bluemira.power_cycle.errors import PowerCycleABCError, PowerCycleLoadABCError
-from bluemira.power_cycle.tools import (
-    FormattedDict,
-    copy_dict_without_key,
-    unique_and_sorted_vector,
-    unnest_list,
-    validate_list,
-    validate_nonnegative,
-    validate_vector,
-)
+from bluemira.power_cycle.errors import PowerCycleError
 
 
-class PowerCycleABC(ABC):
+class ModuleType(Enum):
+    NONE = auto()
+    EQUILIBRIA = auto()
+    PUMPING = auto()
+
+
+class ModuleDescriptor:
+    """Module Descriptor for use with dataclasses"""
+
+    def __set_name__(self, _, name: str):
+        """Set the attribute name from a dataclass"""
+        self._name = "_" + name
+
+    def __get__(self, obj: Any, _) -> str:
+        """Get the module"""
+        return getattr(obj, self._name)
+
+    def __set__(self, obj: Any, value: Union[None, str, ModuleType]):
+        """Set the module"""
+        if value is None:
+            value = ModuleType.NONE
+        elif isinstance(value, str):
+            value = ModuleType[value.upper()]
+
+        setattr(obj, self._name, value)
+
+
+class LoadType(Enum):
+    ACTIVE = auto()
+    REACTIVE = auto()
+
+
+@dataclass
+class PhaseLoadConfig:
+    phase_list: list
+    consumption: bool
+    normalise_list: list
+    powerload_list: list
+
+
+@dataclass
+class BaseConfig:
```

Comment:
The `BaseConfig` class should be renamed to something like `SystemInputElementConfig` or `SystemInstructionConfig` because it is not an ABC. Also, it should be moved to `manager.py` because it is only used there.

Reply:
Same comment about dependency loops